### PR TITLE
Refactor user dialogs

### LINKED ISSUE #4882: Add to New Pool button cancels the dialog in Users page
1. Click on Users from side menu 
2. On the Users list table click on `view` link to open the individual user page 
3. Click on "Add user to Pool" link 
4. Below dialog will appear and click on "Add to new pool"

5. Observe it works as same as "Cancel and go back" link 

### A/C
I believe the objective here is to create a new pool & associate the user with it. 

Question  
Should we open the pool creation page ?
How do we navigate back to this dialog ?

### PR DESCRIPTION
**Pull request by petertgiles**
# :wave: Introduction
The dialogs on the `/admin/users/:userId` page weren't working properly.  The forms were built using raw input partials and managing their own validation and navigation.  I factored them out into their own files and refactored them to use newer form components.

# :test_tube: Testing
image
- [ ] Interact with each of the three dialogs
  - [ ] Start off closed
  - [ ] Can open them by clicking triggers
  - [ ] Can cancel open dialogs
  - [ ] Client-side form validation works (and does not close dialog)
  - [ ] Server-side errors are displayed (and does not close dialog)
    - Try stopping the php container for easy server-side errors
  - [ ] Successful api mutation displays success message and closes the dialog
- [ ] Looks good in French

# :robot: Robot Stuff
Closes #4882

### CODE REVIEW

File: frontend/admin/src/js/components/user/GeneralInformationTab/AddToPoolDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+export const AddToPoolDialog: React.FC<AddToPoolDialogProps> = ({
+  user,
+  pools,
+}) => {
+  const intl = useIntl();
+  const [open, setOpen] = React.useState(false);
+  const methods = useForm<FormValues>();
+
+  const [, executeMutation] = useCreatePoolCandidateMutation();
```
Comment (esizer) [inline frontend/admin/src/js/components/user/GeneralInformationTab/AddToPoolDialog.tsx:40]: Could we pop off `fetching` here and use it to disable the button and update the text content so the user knows something is happening?

> **Note**
> Applies to all three forms

```tsx
const [{ fetching }, executeMutation] = useCreatePoolCandidateMutation();

...

<Button disabled={fetching}>
  {fetching ? intl.formatMessage({
    defaultMessage: "Saving...",
  }) : intl.formatMessage({
     defaultMessage: "Add to new pool",
  })}
</Button>
```

**Review by esizer — CHANGES_REQUESTED** [review]
Oh wow, this is awesome! I just had one small suggestion to improve the UX a bit :)

**Review by esizer — APPROVED** [review]
Amazing 🤩

**Comment by yonikid15** [comment]
😆 Eric beat me to it

**Pull request merged**